Add auction series (#2183)
near/docs: docs/3.tutorials/auction/1.2-testing.md

:::

---

## Account Creation

The first thing our test does is to create multiple accounts with 10 $NEAR tokens each, and deploy the contract into one of them.

<Tabs groupId="code-tabs">
    <TabItem value="js" label="🌐 JavaScript">
        <Github fname="main.ava.js" language="javascript"
                url="https://github.com/near-examples/auctions-tutorial/blob/main/contract-ts/01-basic-auction/sandbox-test/main.ava.js#L12"
                start="12" end="22" />
        
        To deploy the contract, we pass the path to the compiled WASM contract as an argument to the test in `package.json`. Indeed, when executing `npm run test`, the command will first compile the contract and then run the tests.

    </TabItem>
    <TabItem value="rust" label="🦀 Rust">
        <Github fname="test_basics.rs" language="rust"
                url="https://github.com/near-examples/auctions-tutorial/blob/main/contract-rs/01-basic-auction/tests/test_basics.rs#L10"
                start="17" end="28" />

        Notice that the sandbox compiles the code itself, so we do not need to pre-compile the contract before running the tests.
    </TabItem>
</Tabs>

---

## Contract Initialization

> To initialize the contract the contract's account calls itself, invoking the `init` function with an `end_time` set to 60 seconds in the future.

Reviewer: ```suggestion
To initialize the contract's account calls itself, invoking the `init` function with an `end_time` set to 60 seconds in the future.
```

        <Github fname="test_basics.rs" language="rust"
                url="https://github.com/near-examples/auctions-tutorial/blob/main/contract-rs/01-basic-auction/tests/test_basics.rs"
                start="85" end="95" />

    </TabItem>
</Tabs>

---

## Executing the tests 

Now that we understand what we are testing, let's go ahead and run the tests!

<Tabs groupId="code-tabs">

    <TabItem value="js" label="🌐 JavaScript">

        ```bash
        # if you haven't already, install the dependencies
        npm install

        # run the tests
        npm run test 
        ```

    </TabItem>

    <TabItem value="rust" label="🦀 Rust">

        ```bash
        cargo test
        ```

    </TabItem>

</Tabs>

All tests should pass, and you should see the output of the tests in the console. If you see any errors, please contact us in the [NEAR Discord](https://near.chat) or through [Telegram](https://t.me/neardev) and we'll help you out!

---

## Conclusion 

In this part of the tutorial, we've seen how to use our sandbox testing environment to test the contract. We've tested the contract's initialization, biding, and time advancement.

> You are now ready to move to the next section, in which we will deploy the contract to the testnet and interact with it through the CLI.

Reviewer: ```suggestion
You are now ready to move to the next section, where we will deploy the contract to `testnet` and interact with it through the CLI.
```
